functions.py

```diff
@@ -122,11 +122,17 @@ def get_jira_dict(ticket_ids, config):
 
 		# get ticket info from jira API
 		try:
+
+			# initialize connection if it has not yet been done (either first iteration or previously failed)
+			if jira is None:
+				jira = JIRA(auth=auth, options=options)
```

Comment:
Hmm, I would try and move this kind of logic to the outside of the loop. You already do the leg work with building up the `auth` and `options` structures, so I would just try and make the call to the Jira constructor there.

Reply:
I guess in the case where `ticket_ids` is empty (which doesn't seem to be very likely), you waste making this call, but it still makes sense to me to try and stray away from making this call in a loop iteration.

Reply:
The issue here lies in the handling - we don't want failure to connect to the Jira API to end the program right here - we want the functionally present in the `except` and `finally` clauses within the loop, such that the function can return a valid `tickets` dictionary

Reply:
I guess that's not something I previously considered. If that's the case, I would investigate passing an initialized Jira constructor to this helper function, and having the actual call site (or driver function) make this call to the Jira constructor. That way, this helper is only managing a single functionality - iterate over the list of the `@ticket_ids` parameter and return this built-up `tickets` dictionary. I would assume you would still need to wrap the `jira.issue(...)` call in a try/except block.

Reply:
And if what I suggested isn't something that's feasible without making many modifications, or it's out of scope for this PR, I would probably still suggest implementing some sort of polling functionality that tries making this call to the Jira constructor, `x` number of times, before giving up and returning an error.

Reply:
This is out of scope, but we can explore such architectural changes at a later date.

Reply:
That being said, I would welcome an RFE further exploring this, similar to what @EliadCohen did for Bugzilla in #53